src/vmime/net/imap/IMAPConnection.cpp

```diff
@@ -397,7 +409,7 @@ void IMAPConnection::authenticateSASL() {
 
 		shared_ptr <IMAPCommand> authCmd;
 
-		if (saslSession->getMechanism()->hasInitialResponse()) {
+		if (saslIR && saslSession->getMechanism()->hasInitialResponse()) {
```

Comment:
Or we could simply use hasCapability() here, maybe more concise?

`if (hasCapability("SASL-IR") && saslSession->getMechanism()->hasInitialResponse()) {`